### PR TITLE
Hamzahsn/feat/show modal only when api does not endpoint exists

### PR DESCRIPTION
**Pull request by hamzahsn**
This PR...

## Changes

- This PR is about rendering the modal where user specify the api endpoint if no API were specified while opening the dashboard or no API found in localStorage.

## Fixes

-

## How to test it

-

## screenshots

- 

## Checklist

- [x ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Comment by hamzahsn** [comment]
When using this feature with staged env on Netlify - API does not work - CORS problem, needs to check that with @fog1985 and @nicufk

File: src/components/atoms/Modal/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        </Typography>
+        <StyledFormContainer>
+          <LabelInput
+            id="url"
+            name="url"
+            onChange={(event: React.ChangeEvent<HTMLInputElement>) => handleChangeApiEndpoint(event, 'apiEndpoint')}
+            defaultValue={apiEndpoint.apiEndpoint}
+          />
+          <Button type="submit" disabled={!validUrl} disableFilter>
+            Get tests
```
Comment (olensmar) [inline src/components/atoms/Modal/Modal.tsx:93]: I think we said "Get Results" for this button?

File: src/components/atoms/Modal/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        <Typography variant="secondary">
+          The endpoint needs to be accessible from your browser and will be used to retrieve test results only.
+        </Typography>
+        <StyledFormContainer>
+          <LabelInput
+            id="url"
+            name="url"
+            onChange={(event: React.ChangeEvent<HTMLInputElement>) => handleChangeApiEndpoint(event, 'apiEndpoint')}
+            defaultValue={apiEndpoint.apiEndpoint}
+          />
```
Comment (olensmar) [inline src/components/atoms/Modal/Modal.tsx:91]: I noticed in the current demo dashboard that selecting an endpoint provided by the browser-history does not enable the button, you have to make an actual change with the keyboard  - is this still an issue here?

File: src/components/atoms/Modal/Modal.tsx (hunk 1 above)
Comment (hamzahsn) [reply src/components/atoms/Modal/Modal.tsx:93]: You're right, but I have updated this one in another PR which I'm almost done from it https://github.com/kubeshop/kubtest-dashboard/pull/44

File: src/components/atoms/Modal/Modal.tsx (hunk 2 above)
Comment (hamzahsn) [reply src/components/atoms/Modal/Modal.tsx:91]: this one I have also tried to handle it in the same PR which mentioned above.

**Review by olensmar — APPROVED** [review]
ok for now - let's see how it all comes together!

**Pull request merged**